### PR TITLE
Stop using THLongStorage for sizes/strides, remove THLongStorageView.

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

**Comment by gchanan** [comment]
This is stacked on top of https://github.com/pytorch/pytorch/pull/10209, to review look at https://github.com/pytorch/pytorch/pull/10219/files/ad94ee36272b5f6a2d631974618a87f95afac159..HEAD.

File: aten/src/ATen/THSizeStrideCompat.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// 1) first class scalar support (for sizes)
+// 2) differentiating between nullptr/non-nullptr strides (the former "infers" strides).
+
+namespace at {
+
+static inline at::IntList get_intlist_size_th(IntList sizes) {
+  static int64_t one = 1;
+  if (sizes.size() == 0) {
+    // fake scalar
+    return IntList(&one, 1);
```
Comment (ezyang) [inline aten/src/ATen/THSizeStrideCompat.h]: Classy :)

File: aten/src/TH/THTensor.cpp [hunk 2]
```diff
@@ -14,6 +14,128 @@ void THTensor_free(THTensor *self)
...
+                        stride_.data());
+}
+
+void THTensor_setStorageNd(THTensor *self, THStorage *storage, ptrdiff_t storageOffset, int nDimension, const int64_t *size, const int64_t *stride)
+{
+  /* storage */
+  if(THTensor_getStoragePtr(self) != storage)
+  {
+    if (!THTensor_getStoragePtr(self)) {
+      THError("Tensor: invalid null storage");
```
Comment (ezyang) [inline aten/src/TH/THTensor.cpp]: Apropos of nothing... "Call GetMutableData first!"

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**